### ISSUE TITLE
AugmentedDiff.get_state() returns False and/or raises ReadTimeout exception on Windows 7, Python 3.7.3

### ISSUE THREAD

**Issue by vitale232** [mvexel/osmdiff#11]
## Issue Description
I'm trying to follow the simple `AugmentedDiff` example from the readme. During my first attempt, the `AugmentedDiff.get_state()` method returned False. On some subsequent attempts, the method raises an exception from the `requests` module:

> `>>> a.get_state()`
>`requests.exceptions.ReadTimeout: HTTPConnectionPool(host='overpass-api.de', port=80): Read timed out. (read timeout=5)`

## Environment
Windows 7 (64-bit)
Python 3.7.3 (Anaconda Distribution)
conda 4.7.5
osmdiff 0.1.8 (pip install)

## Workaround
The workaround seems to be using `requests` to get the augmented diff sequence number, then manually assign it to the `AugmentedDiff.sequence_number` property.
```python
>>> from osmdiff import AugmentedDiff
>>> import requests
>>>
>>> a = AugmentedDiff()
>>> response = requests.get(a.base_url  + '/augmented_diff_status')
>>> a.sequence_number = int(response.text)
>>> a.sequence_number
3577361
>>> a.retrieve()
>>> a
AugmentedDiff (1534 created, 343 modified, 125 deleted)
>>>
```

I believe the use of `os.path.join` in the `AugmentedDiff.get_state` method is producing an invalid URL on Windows.

**Comment by mvexel** [comment]
Hmm your assessment may very well be correct, I haven't tested on Windows. I'll see if I can get a chance to do that, I do have a windows (10) machine but would need to install a python developer environment.. If you can positively confirm that `os.path.join` is the culprit here, that would help :)

**Comment by vitale232** [comment]
Here's some screenshots to help verify... The first screenshot is using Python3.9 in Win10. The second screenshot is using Python3.8 in Ubuntu, via Windows Subsytem for Linux 2:

![image](https://user-images.githubusercontent.com/8504254/114751995-d7827780-9d23-11eb-9259-d34d6b955277.png)

![image](https://user-images.githubusercontent.com/8504254/114752053-e8cb8400-9d23-11eb-99d0-dfff67ada933.png)

Hope that helps, and thanks for the software! :-)

**Comment by mvexel** [comment]
OK that gives me something to work with! Obviously I am abusing `os.path.join` to construct a URL when it is meant for file URIs which are OS-dependent. Let me see what I can do here!

**Comment by mvexel** [comment]
It looks like I need to replace `os.path.join` with [`urllib.urlparse.urljoin`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin) but I don't have an easy way to test the result on Windows. Anyone able to help here?

**Comment by docentYT** [comment]
#21

**Comment by mvexel** [comment]
This should be solved with #28 and the published 0.4. Lmk if there's still issues with it.
